public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx

```diff
@@ -229,7 +229,8 @@ export function getVisualizationOptions2(props: OptionPaneRenderProps2): Options
         render: function renderEditor() {
           const onChange = (v: unknown) => {
             panel.onFieldConfigChange(
-              updateDefaultFieldConfigValue(currentFieldConfig, fieldOption.path, v, fieldOption.isCustom)
+              updateDefaultFieldConfigValue(currentFieldConfig, fieldOption.path, v, fieldOption.isCustom),
+              true
```

Comment:
@dprokop I had to switch replace: true here, as the "replace" false logic does handle emtpy arrays a bit strangely. 

If you pass an empty array for links for example, and let replace be false then the merge operation will add back the current links so you cannot clear/empty an array when your passing false (or leaving out replace param). So in main right now you cannot remove an item from an array (links for example, or value mappings). 
 
Feels like this merge behavior is not what we intended? 
https://github.com/grafana/scenes/blob/main/packages/scenes/src/components/VizPanel/VizPanel.tsx#L259

Reply:
Indeed, the `replace` is aimed primarily at changing the field config while defaulting things that are not passed in the partial config. Maybe there's a better way to handle that.

Reply:
@dprokop created an issue in scenes repo to investigate it at some later time :) https://github.com/grafana/scenes/issues/607